### PR TITLE
Implement socket callback

### PR DESCRIPTION
**Pull request by michaelarnauts**
* Documentation improved
* Implement socket callback
* Implement settings

TODO:
* Update documentation on socket callback

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,17 +20,21 @@ know the endpoints for the Channel and EPG data.
 {
     "version": 1,
     "channels": "plugin://plugin.video.example/iptv/channels?output=$FILE",
-    "epg": "plugin://plugin.video.example/iptv/epg?output=$FILE"
+    "channels_format": "json",
```
Comment (dagwieers) [inline README.md:23]: Can we not change `channels` into `channels_url`? So we can also add the (static) channels listing in this file as `channels`?

That would leave us with just one file for dynamic _and_ static use-cases.

Comment (michaelarnauts) [reply README.md:23]: That would use two keys then, this causes more confusion when both keys are specified to know what has priority. You could use the static listing in 'channels' fine, or am I missing something?

Comment (michaelarnauts) [reply README.md:23]: We do the check here: https://github.com/add-ons/service.iptv.manager/blob/763657b57145c1e28a3b52923488ce427b1694ca/resources/lib/modules/addon.py#L118

If it's not an url (plugin or http), we try it as a file.

Comment (dagwieers) [reply README.md:23]: I didn't know there was a shortage on JSON keys :upside_down_face: 

It is generally frowned upon in JSON (and any schemaless format for that matter) to have a value that can be both a string and a dict. Another possibility is:
```json
{
  "channels": {
    "format": "json",
    "listing": {
      ...
    },
    "url": "plugin://plugin.video.example/iptv/channels?output=$FILE"
  },
  "epg": {
    "format": "json",
    "url": "plugin://plugin.video.example/iptv/epg?output=$FILE"
  },
  "version": 1,
}
```

I also think we should make an _official_ standard for our JSON format for channels and epg, and give it a name (jsontv and jsonepg?). If anything already exists, we could collaborate, but I haven't found anything conclusive.

Comment (dagwieers) [reply README.md:23]: Hmm, I guess wrt. implementation, leaving the listing out of this JSON makes more sense. I updated my example above. And the Design wiki page.

We should support `special://`, but supporting `file:///` seems useless.

Comment (dagwieers) [reply README.md:23]: Can we not detect the format instead?

I would still prefer to use *settings.xml* so we don't have to access these files from add-on directories. I would expect this to be cached by Kodi and therefore faster than individual file accesses. If not, we have to make sure we get an exception from Kodi devs first.

File: addon.xml [hunk 2]
```diff
@@ -5,6 +5,9 @@
         <import addon="script.module.dateutil" version="2.6.0"/>
     </requires>
     <extension point="xbmc.service" library="service.py"/>
+	<extension point="xbmc.python.script" library="functions.py">
```
Comment (michaelarnauts) [inline addon.xml:8]: @dagwieers weet jij of dit moet? Lijkt ook zonder te werken...

File: resources/lib/functions.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+from resources.lib import kodilogging, kodiutils
+from resources.lib.modules.iptvsimple import IptvSimple
+
+kodilogging.config()
+_LOGGER = logging.getLogger(__name__)
+
+
+# TODO: throws the error:
+# RuntimeError: No valid addon id could be obtained. None was passed and the script wasn't executed in a normal xbmc manner.
```
Comment (michaelarnauts) [inline resources/lib/functions.py:16]: Note to self: Zou moeten kunnen als de eigen addon id wordt meegegeven met Addon()

**Comment by michaelarnauts** [comment]
I think this can better be merged. Future changes can happen in additional PR's.

**Pull request merged**